packages/network-controller/src/NetworkController.test.ts

```diff
@@ -218,7 +218,9 @@ describe('NetworkController', () => {
                     pollingInterval: 12000,
                   },
                 });
-                expect(controller.provider).toBe(fakeMetamaskProvider);
+                expect(controller.getProviderAndBlockTracker().provider).toBe(
```

Comment:
As I wrote in Slack, this assertion fails because `getProviderAndBlockTracker` now returns a proxy around the provider instead of the provider itself.

Unfortunately, there doesn't seem to be a way to get access to the target that the proxy is wrapping, otherwise we could say something like:

``` javascript
expect(controller.getProviderAndBlockTracker().provider.target).toBe(
  fakeMetamaskProvider
);
```

Instead, I think what we ought to do is mock an arbitrary request on the provider (it doesn't matter what the RPC method is called) and then test that we can use the provider to request this RPC method and that the method returns what we set.

We can create the request mock by passing an array to `buildFakeMetamaskProvider`. You could do something like this:

``` javascript
const mockResult = 'the result';
const fakeMetamaskProvider = buildFakeMetamaskProvider([
  {
    request: {
      method: 'test',
    },
    response: {
      result: mockResult
    }
  }
]);
```

Then, instead of testing what the provider is, you can call `sendAsync` on the EthQuery object (obtained from the controller by calling the `NetworkController:getEthQuery` action) to make a request using this method:

``` javascript
const ethQuery = messenger.call('NetworkController:getEthQuery');
const promisifiedSendAsync = promisify(ethQuery.sendAsync).bind(ethQuery);
const expectedResult = await promisifiedSendAsync({ method: 'test' });
expect(actualResult).toStrictEqual(mockResult);
```

You can see an example of this pattern in the unit tests I've written for the extension NetworkController here: https://github.com/MetaMask/metamask-extension/blob/335219c5f03dbadaeb2c4b8eadeb12d7ecee481b/app/scripts/controllers/network/network-controller.test.js#L556